src/main/java/eu/europeana/metis/sandbox/service/workflow/harvest/FileHarvestTarget.java

```diff
@@ -0,0 +1,18 @@
+package eu.europeana.metis.sandbox.service.workflow.harvest;
+
+import eu.europeana.metis.sandbox.common.FileType;
+import lombok.Value;
+
+/**
+ * Represents a target file to be harvested in the system.
+ * <p>
+ * The FileHarvestTarget class encapsulates the metadata and content of a file to be used for harvesting operations. It includes
+ * the file's name, type, and raw byte content.
+ */
+@Value
+public final class FileHarvestTarget {
```

Comment:
suggestion just as an idea use record instead? why a class?

Reply:
Aaah yes! I had it as record but sonar started complaining about using the byte array in a custom equals and hashCode and toString. So I opted out to Lombok which also doesn't use it but avoids the sonar warning.
In any case, I reverted to record and added an explicit SuppressWarning. Not sure if there is an other way to do this tbh.